pkg/capabilities/capabilities_test.go

```diff
@@ -48,10 +42,57 @@ func TestGet(t *testing.T) {
 			HostNetworkSources: []string{"A", "B"},
 		},
 	}
-	SetForTests(cap)
+	ResetForTest(&cap)
 
 	res = Get()
 	if !reflect.DeepEqual(cap, res) {
 		t.Fatalf("expected Capabilities: %#v , got a different: %#v", cap, res)
 	}
 }
+func TestSetup(t *testing.T) {
+	testCases := []struct {
+		name                     string
+		allowPrivileged          bool
+		perConnectionBytesPerSec int64
+		expectedCapabilities     Capabilities
+	}{
+		{
+			name:                     "AllowPrivileged true with bandwidth limit",
+			allowPrivileged:          true,
+			perConnectionBytesPerSec: 1024,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        true,
+				PerConnectionBandwidthLimitBytesPerSec: 1024,
+			},
+		},
+		{
+			name:                     "AllowPrivileged false with higher bandwidth limit",
+			allowPrivileged:          false,
+			perConnectionBytesPerSec: 2048,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        false,
+				PerConnectionBandwidthLimitBytesPerSec: 2048,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer ResetForTest(nil)
```

Comment:
Just wondering if this call should be moved up to avoid resetting 2 times?

Reply:
I might be misunderstanding what you mean, but I believe it's necessary to keep the `ResetForTest(nil)` call inside each test case to ensure each test starts with a clean state. This is confirmed by the fact that if I take it out, the second test case fails due to the `Once` not being reset properly and the subsequent call to `capabilities.once.Do` being ignored.

Reply:
My bad. I thought that `ResetForTest` will be called after the loop. I didn't notice that it's called in the function.